mm2src/coins/eth/fee_estimation/eip1559/block_native.rs

```diff
@@ -0,0 +1,158 @@
+use super::{EstimationSource, FeePerGasEstimated, FeePerGasLevel, FEE_PER_GAS_LEVELS};
+use crate::eth::{Web3RpcError, Web3RpcResult};
+use crate::{wei_from_gwei_decimal, NumConversError};
+use mm2_err_handle::mm_error::MmError;
+use mm2_err_handle::prelude::*;
+use mm2_net::transport::slurp_url_with_headers;
+use mm2_number::BigDecimal;
+
+use http::StatusCode;
+use serde_json::{self as json};
+use std::convert::TryFrom;
+use std::convert::TryInto;
+use url::Url;
+
+lazy_static! {
+    /// API key for testing
+    static ref BLOCKNATIVE_GAS_API_AUTH_TEST: String = std::env::var("BLOCKNATIVE_GAS_API_AUTH_TEST").unwrap_or_default();
+}
+
+#[allow(dead_code)]
+#[derive(Clone, Debug, Deserialize)]
+pub(crate) struct BlocknativeBlockPrices {
+    #[serde(rename = "blockNumber")]
+    pub block_number: u32,
+    #[serde(rename = "estimatedTransactionCount")]
+    pub estimated_transaction_count: u32,
+    #[serde(rename = "baseFeePerGas")]
+    pub base_fee_per_gas: BigDecimal,
+    #[serde(rename = "estimatedPrices")]
+    pub estimated_prices: Vec<BlocknativeEstimatedPrices>,
+}
+
```

Comment:
```suggestion
 #[allow(dead_code)]
 #[derive(Clone, Debug, Deserialize)]
 #[serde(rename_all = "camelCase")
 pub(crate) struct BlocknativeBlockPrices {
     pub block_number: u32,
     pub estimated_transaction_count: u32,
     pub base_fee_per_gas: BigDecimal,
     pub estimated_prices: Vec<BlocknativeEstimatedPrices>,
 }
```

applies to other enums below I guess

Reply:
backlogged in https://github.com/KomodoPlatform/komodo-defi-framework/issues/2207